pycloudlib/azure/cloud.py

```diff
@@ -615,15 +641,26 @@ def _create_virtual_machine(
         if vm_params:
             update_nested(params, vm_params)
         self._log.debug("Creating Azure virtual machine: %s", name)
-        vm_poller = (
-            self.compute_client.virtual_machines.begin_create_or_update(
-                self.resource_group.name,
-                name,
-                params,
+        try:
+            vm_poller = (
+                self.compute_client.virtual_machines.begin_create_or_update(
+                    self.resource_group.name,
+                    name,
+                    params,
+                )
             )
-        )
-
-        return vm_poller.result()
+            vm_poller.wait(provisioning_timeout)
+            if not vm_poller.done():
+                raise PycloudlibTimeoutError(
+                    "Virtual machine creation timed out."
+                )
+            return vm_poller.result()
+        except HttpResponseError as e:
+            err_code = e.error.code
+            err_msg = e.error.message
+            raise PycloudlibTimeoutError(
```

Comment:
I think we should use another exception here, as we do not for sure that's the case.

Reply:
Agreed. Thanks for the suggestion.